Applications/Subgraph/jni/main.cpp

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2024 Eunju Yang <ej.yang@samsung.com>
+ *
+ * @file   main.cpp
+ * @date   27 Dec 2024
+ * @brief  Test Application for subgraph weight sharing
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Eunju Yang <ej.yang@samsung.com>
+ * @bug    No known bugs except for NYI items
+ */
+
+#include <cifar_dataloader.h>
+#include <layer.h>
+#include <model.h>
+#include <optimizer.h>
+
+#include <array>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include <model_util.h>
+
+using LayerHandle = std::shared_ptr<ml::train::Layer>;
+using ModelHandle = std::unique_ptr<ml::train::Model>;
+using UserDataType = std::unique_ptr<nntrainer::util::DataLoader>;
+
+/**
+ * @brief tain data callback
+ */
+int trainData_cb(float **input, float **label, bool *last, void *user_data) {
```

Comment:
It seems that you are not using a data loader, is this necessary?

Reply:
It's my bad 😅 I fix the redundant codes. Thanks :)